Xamarin.Essentials/TextToSpeech/TextToSpeech.uwp.cs

```diff
@@ -0,0 +1,124 @@
+
+                    player.MediaEnded += PlayerMediaEnded;
+                    player.Source = MediaSource.CreateFromStream(stream, stream.ContentType);
+                    player.Play();
+
+                    void OnCancel()
+                    {
+                        player.PlaybackSession.PlaybackRate = 0;
+                        tcs.TrySetResult(true);
+                    }
+
+                    using (cancelToken.Register(OnCancel))
+                    {
+                        await tcs.Task;
+                    }
+
+                    player.MediaEnded -= PlayerMediaEnded;
+                    player.Dispose();
+
+                    void PlayerMediaEnded(MediaPlayer sender, object args)
+                    {
+                        tcs.TrySetResult(true);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine("Unable to playback stream: " + ex);
+                }
+            }
+            finally
+            {
+                if (semaphore.CurrentCount == 0)
+                    semaphore.Release();
+            }
+        }
+
+        static string GetSpeakParametersSSMLProsody(string text, SpeakSettings settings)
+        {
+            var volume = "default";
+            var pitch = "default";
+            var rate = "default";
+
+            // Look for the specified language, otherwise the default voice
+            var locale = settings?.Locale.Language ?? SpeechSynthesizer.DefaultVoice.Language;
+
+            if (settings?.Volume.HasValue ?? false)
+                volume = (settings.Volume.Value * 100f).ToString();
+
+            if (settings?.Pitch.HasValue ?? false)
+                pitch = ProsodyPitch(settings.Pitch);
+
+            if (settings?.SpeakRate.HasValue ?? false)
+                rate = settings.SpeakRate.Value.ToString();
+
+            // SSML generation
+            var ssml = new StringBuilder();
+            ssml.AppendLine($"<speak version='1.0' xmlns='http://www.w3.org/2001/10/synthesis' xml:lang='{locale}'>");
+            ssml.AppendLine($"<prosody pitch='{pitch}' rate='{rate}' volume='{volume}'>{text}</prosody> ");
+            ssml.AppendLine($"</speak>");
+
+            return ssml.ToString();
+        }
+
+        static string ProsodyPitch(float? pitch)
+        {
+            if (!pitch.HasValue)
+                return "default";
+
+            if (pitch.Value <= 0.25f)
+                return "x-low";
+            else if (pitch.Value > 0.25f && pitch.Value <= 0.75f)
+                return "low";
+            else if (pitch.Value > 0.75f && pitch.Value <= 1.25f)
+                return "medium";
+            else if (pitch.Value > 1.25f && pitch.Value <= 1.75f)
+                return "high";
+            else if (pitch.Value > 1.75f)
+                return "x-high";
+
+            return "default";
+        }
```

Comment:
I tried to change these values, but my SurfaceBook makes no audible changes - is this just me?

Reply:
i think i had broke it before maybe? we can re-test....